applications/openshift/api-server/api_server_insecure_port/rule.yml

```diff
@@ -42,8 +42,27 @@ ocil: |-
     Run the following command on the master node(s):
 {{%- if product == "ocp4" %}}
     <pre>$ oc get configmap config -n openshift-kube-apiserver -ojson | jq -r '.data["config.yaml"]' | jq '.apiServerArguments["insecure-port"]'</pre>
-    The output should return <pre>false</pre>.
+    The output should return <pre>0</pre>.
 {{% else %}}
     <pre>$ sudo grep -A2 insecure-port /etc/origin/master/master-config.yaml</pre>
-{{%- endif %}}
     There should be no output returned.
+{{%- endif %}}
+
+{{%- if product == "ocp4" %}}
+warnings:
+- general: |-
+    {{{ openshift_cluster_setting("/api/v1/namespaces/openshift-kube-apiserver/configmaps/config") | indent(4) }}}
+{{%- endif %}}
+
+# This will check that insecure-port is "0" or 0.
+template:
+  name: yamlfile_value
+  vars:
+    ocp_data: "true"
+    entity_check: "all"
+    filepath: /api/v1/namespaces/openshift-kube-apiserver/configmaps/config
+    yamlpath: '.data["config.yaml"]'
+    values:
+    - value: '"insecure-port":\[["|]0["|]\]'
```

Comment:
What is the intention behind the `["|]`? Would an integer value be really quoted by pipes instead of quotes?

Reply:
regex format is weird. Sorry for that. The format intended here is as follows: `[<value 1>|<value 2>]` -> `<value 1> or <value 2>`. in this case `["|]` will check for either `"` or nothing. This can match either `"0"` or `0`.

Reply:
I am pretty sure that it's not the case - OVAL defines that the PCRE family of regexes should be used, and it can be tested e.g. on https://regex101.com You can see that the pipe inside square brackets is taken literally, but you can express that using `"?`.

Reply:
@matejak thanks for the review! I'll change it then